x/auth/ante/sigverify.go

```diff
@@ -255,23 +155,69 @@ func (svd SigVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 		return ctx, errorsmod.Wrapf(sdkerrors.ErrUnauthorized, "invalid number of signer;  expected: %d, got %d", len(signers), len(signatures))
 	}
 
+	pubKeys, err := sigTx.GetPubKeys()
+	if err != nil {
+		return ctx, err
+	}
+
+	// NOTE: the tx_wrapper implementation returns nil, in case the pubkey is not populated.
+	// so we can always expect the pubkey of the signer to be at the same index as the signer
+	// itself. If this does not work, it's a failure in the implementation of the interface.
+	// we're erroring, but most likely we should be panicking.
```

Comment:
Why would we panic? Especially in an ante-handler?

Reply:
this is a breach of the sdk.Tx interface implementation

Reply:
a panic would be caught and treated as an error anyway, so a panic would be futile.